test/new-e2e/pkg/utils/e2e/client/agentclientparams/agent_client_params.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package agentclientparams implements function parameters for [e2e.Agent]
+package agentclientparams
+
+// Params defines the parameters for the Agent client.
+// The Params configuration uses the [Functional options pattern].
+//
+// The available options are:
+//   - [WithSkipWaitForAgentReady]
+//
+// [Functional options pattern]: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+type Params struct {
+	ShouldWaitForReady bool
+}
+
+// Option alias to a functional option changing a given Params instance
+type Option func(*Params)
+
+// NewParams creates a new instance of Agent client params
+// default ShouldWaitForReady: true
+func NewParams(options ...Option) *Params {
+	p := &Params{
+		ShouldWaitForReady: true,
```

Comment:
I think either `ShouldWaitForReady` should be `false` by default or https://github.com/DataDog/datadog-agent/blob/79cd7b4739d51ae251c7dc5831e0db52f3d02d0c/test/new-e2e/pkg/utils/e2e/client/agent_cmd_runner.go#L39-L43 should be removed. What do you think?

Reply:
I think the call in command can be removed now, can do it within this PR or in a follow up to clean up code